game-core/src/main/java/games/strategy/engine/data/export/GameDataExporter.java

```diff
@@ -267,409 +218,356 @@ private void relationshipInitialize(final GameData data) {
+                .map(GameDataExporter::printAttachments)
+                .collect(Collectors.toList()))
+        .build();
   }
 
-  private void playerProduction(final GameData data) {
-    for (final GamePlayer player : data.getPlayerList()) {
-      try {
-        final String playerfrontier = player.getProductionFrontier().getName();
-        final String playername = player.getName();
-        xmlfile
-            .append("        <playerProduction player=\"")
-            .append(playername)
-            .append("\" frontier=\"")
-            .append(playerfrontier)
-            .append("\"/>\n");
-      } catch (final NullPointerException npe) {
-        // neutral?
-      }
+  private static List<AttachmentList.Attachment.Option> printAttachmentOptionsBasedOnOriginalXml(
+      final List<Tuple<String, String>> attachmentPlusValues) {
+    final List<AttachmentList.Attachment.Option> options = new ArrayList<>();
+    for (final Tuple<String, String> current : attachmentPlusValues) {
+      options.add(
+          AttachmentList.Attachment.Option.builder()
+              .name(current.getFirst())
+              .value(current.getSecond())
+              .build());
     }
+    return options;
   }
 
-  private void productionFrontiers(final GameData data) {
-    for (final String frontierName :
-        data.getProductionFrontierList().getProductionFrontierNames()) {
-      final ProductionFrontier frontier =
-          data.getProductionFrontierList().getProductionFrontier(frontierName);
-      xmlfile.append("\n");
-      xmlfile
-          .append("        <productionFrontier name=\"")
-          .append(frontier.getName())
-          .append("\">\n");
-      for (final ProductionRule rule : frontier.getRules()) {
-        xmlfile
-            .append("            <frontierRules name=\"")
-            .append(rule.getName())
-            .append("\"/>\n");
-      }
-      xmlfile.append("        </productionFrontier>\n");
-    }
-    xmlfile.append("\n");
-  }
-
-  private void productionRules(final GameData data) {
-    for (final ProductionRule pr : data.getProductionRuleList().getProductionRules()) {
-      xmlfile.append("        <productionRule name=\"").append(pr.getName()).append("\">\n");
-      for (final Resource cost : pr.getCosts().keySet()) {
-        xmlfile
-            .append("            <cost resource=\"")
-            .append(cost.getName())
-            .append("\" quantity=\"")
-            .append(pr.getCosts().getInt(cost))
-            .append("\"/>\n");
-      }
-      for (final NamedAttachable result : pr.getResults().keySet()) {
-        xmlfile
-            .append("            <result resourceOrUnit=\"")
-            .append(result.getName())
-            .append("\" quantity=\"")
-            .append(pr.getResults().getInt(result))
-            .append("\"/>\n");
-      }
-      xmlfile.append("        </productionRule>\n");
-    }
-  }
+  private AttachmentList.Attachment printAttachments(
```

Comment:
Codacy found an issue: [The method 'printAttachments(Tuple)' has an NPath complexity of 256, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6193173)